ansible_rulebook/auth.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright 2023 Red Hat, Inc.
```

Comment:
@bzwei another suggestion for filename would be tokens or token helpers. auth has different meaning people would be expecting to see authentication classes or methods in that file.

Reply:
@bzwei I think this file is making an assumption that the token refresh follows the simplejwt model and end points, if we change the backend auth method on the eda-server to something else the ansible-rulebook would have to change. Should we have a field that defines which Auth refresh token method is in play. Simple JWT, OAuth? So maybe this code could adjust itself.

Reply:
@mkanoor Yes this implementation makes the assumption of simple jwt or alike. The CLI takes corresponding arguments. If we switches to another authentication method, the CLI arguments may be different, and the parameters needed for fetching/renewing tokens can be very different too.